crates/storage/provider/src/providers/database/provider.rs

```diff
@@ -821,25 +823,25 @@ impl<TX: DbTxMut + DbTx> DatabaseProvider<TX> {
     pub fn prune_table_with_range<T: Table>(
         &self,
         keys: impl RangeBounds<T::Key> + Clone + Debug,
-        limit: usize,
+        limit: PruneLimit,
         mut skip_filter: impl FnMut(&TableRow<T>) -> bool,
         mut delete_callback: impl FnMut(TableRow<T>),
     ) -> Result<(usize, bool), DatabaseError> {
+        let start = Instant::now();
+
         let mut cursor = self.tx.cursor_write::<T>()?;
         let mut walker = cursor.walk_range(keys)?;
         let mut deleted = 0;
 
-        if limit != 0 {
-            while let Some(row) = walker.next().transpose()? {
-                if !skip_filter(&row) {
-                    walker.delete_current()?;
-                    deleted += 1;
-                    delete_callback(row);
-                }
+        while let Some(row) = walker.next().transpose()? {
```

Comment:
we should leave the initial limit check because otherwise we'll do a `walker.next()` that's not going to be used later

Reply:
seems to be a bug on main too, will open an issue

https://github.com/paradigmxyz/reth/blob/48b763f88a3cf175ded5d5ea9f99559a59c65087/crates/storage/provider/src/providers/database/provider.rs#L845

Reply:
wait, actually, must have been correct from the start, right? walker only lives in scope of this method, so shouldn't be a problem to skip limit check outside of loop - if limit is reached, then the method returns anyway.